### PR TITLE
Allow notification apps to defer and flush the sending

### PR DESCRIPTION
**Pull request by nickvergessen**
Signed-off-by: Joas Schilling <coding@schilljs.com>

### CODE REVIEW

**Comment by nickvergessen** [comment]
So, let's do this for 20?

File: lib/public/Notification/IDeferrableApp.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ * Interface IDeferrableApp
+ *
+ * @package OCP\Notification
+ * @since 20.0.0
+ */
+interface IDeferrableApp extends IApp {
+	/**
+	 * @since 20.0.0
+	 */
+	public function defer(): void;
```
Comment (ChristophWurst) [inline lib/public/Notification/IDeferrableApp.php:38]: no docs

File: lib/public/Notification/IDeferrableApp.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+	/**
+	 * @since 20.0.0
+	 */
+	public function flush(): void;
```
Comment (ChristophWurst) [inline lib/public/Notification/IDeferrableApp.php:43]: no docs

**Review by ChristophWurst — COMMENTED** [review]
I have no idea what this does :D

**Comment by nickvergessen** [comment]
> I have no idea what this does :D

It allows to interact with notifications and only sends them on flush, which means only 1 connection to the push proxy instead of of everytime. E.g. when a call in Talk is started this means 1 connection instead of 2*participants (we hacked it currently to be already 1 only, but this would be an official API for it)

File: lib/public/Notification/IDeferrableApp.php (hunk 2 above)
Comment (nickvergessen) [reply lib/public/Notification/IDeferrableApp.php:47]: Done

**Review by rullzer — APPROVED** [review]
Yes as discussed already

**Review by MorrisJobke — APPROVED** [review]
Code looks good 👍

**Pull request merged**